data_model/src/transaction.rs

```diff
@@ -330,6 +330,20 @@ impl Transaction for VersionedSignedTransaction {
     }
 }
 
+impl From<ValidTransaction> for SignedTransaction {
+    fn from(source: ValidTransaction) -> Self {
+        let ValidTransaction {
+            payload,
+            signatures,
+        } = source;
```

Comment:
You can (should) do this in the function signature. 

Also, this function should be marked as inline (it's always available thanks to the prelude), and IMO it would be much shorter if you just used `Self {payload: source.payload, signatures: source.signatures.into()`.